lib/iris/analysis/maths.py

```diff
@@ -341,10 +341,16 @@ def divide(cube, other, dim=None, in_place=False):
     _assert_is_cube(cube)
     other_unit = getattr(other, 'units', '1')
     new_unit = cube.units / other_unit
-    try:
-        op = operator.idiv if in_place else operator.div
-    except AttributeError:
-        op = operator.itruediv if in_place else operator.truediv
+    if in_place:
+        op = operator.itruediv
+        # Input `other` may not have a `dtype` attribute.
+        other_dtype_kind = other.dtype.kind if hasattr(other, 'dtype') else 'f'
+        if 'i' in [cube.dtype.kind, other_dtype_kind]:
+            # NumPy ufunc `true_divide` cannot coerce output to int.
+            aemsg = 'Cannot perform inplace division of integer data.'
+            raise ArithmeticError(aemsg)
```

Comment:
@dkillick I think that you only need to check that the `cube.dtype.kind` is always `f` for `itruediv` ... right?